core/lib/zksync_core/src/utils/mod.rs

```diff
@@ -262,4 +271,271 @@ mod tests {
+        fn with_tx_error() -> Self {
+            Self::with_error(EthClientError::EthereumGateway(
+                zksync_types::web3::Error::Unreachable,
+            ))
+        }
+
+        fn with_legacy_contract() -> Self {
+            Self::with_contract_error()
+        }
+
+        fn with_rollup_contract() -> Self {
+            Self::with_retval(vec![ethabi::Token::Uint(U256::zero())])
+        }
+
+        fn with_validium_contract() -> Self {
+            Self::with_retval(vec![ethabi::Token::Uint(U256::one())])
+        }
+    }
+
+    #[async_trait]
+    impl EthInterface for MockEthereumForCommitGenerationMode {
+        async fn get_tx_status(
+            &self,
+            _: H256,
+            _: &'static str,
+        ) -> Result<Option<ExecutedTxStatus>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn send_raw_tx(&self, _: RawTransactionBytes) -> Result<H256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn nonce_at_for_account(
+            &self,
+            _: Address,
+            _: BlockNumber,
+            _: &'static str,
+        ) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn base_fee_history(
+            &self,
+            _: usize,
+            _: usize,
+            _: &'static str,
+        ) -> Result<Vec<u64>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn get_pending_block_base_fee_per_gas(
+            &self,
+            _: &'static str,
+        ) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn get_gas_price(&self, _: &'static str) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn block_number(&self, _: &'static str) -> Result<U64, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn failure_reason(&self, _: H256) -> Result<Option<FailureInfo>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn get_tx(
+            &self,
+            _: H256,
+            _: &'static str,
+        ) -> Result<Option<Transaction>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn tx_receipt(
+            &self,
+            _: H256,
+            _: &'static str,
+        ) -> Result<Option<TransactionReceipt>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn eth_balance(&self, _: H160, _: &'static str) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn call_contract_function(
+            &self,
+            _: ContractCall,
+        ) -> Result<Vec<ethabi::Token>, EthClientError> {
+            let mut error = None;
+            core::mem::swap(&mut *self.error.lock().unwrap(), &mut error);
```

Comment:
Funnily that's roughly how `Option::take` is [implemented under the hood](https://doc.rust-lang.org/src/core/option.rs.html#1688). You should still be able to use `take` with a mutex, but nbd.

Reply:
I tried to use `take` but it requires the inner type to implement `Default`.

Reply:
Though that was with `mem::take`. Maybe `Option::take` works.